include/dxc/WinAdapter.h

```diff
@@ -912,16 +912,60 @@ unsigned int SysStringLen(const BSTR bstrString);
 #define CP_ACP 0
 #define CP_UTF8 65001 // UTF-8 translation.
 
-// Convert Windows codepage value to locale string
-const char *CPToLocale(uint32_t CodePage);
+// RAII style mechanism for setting/unsetting a locale for the specified Windows
+// codepage
+class ScopedLocale {
+  const char *m_locale;
+
+public:
+  explicit ScopedLocale(uint32_t codePage)
+      : m_locale(SetLocaleForCodePage(LC_ALL, codePage)) {}
+  ~ScopedLocale() {
+    if (m_locale != nullptr) {
+      setlocale(LC_ALL, m_locale);
+    }
+  }
+  bool IsSupported() { return (m_locale != nullptr); }
+
+private:
+  // Sets a locale for the specified Windows codepage
+  const char *SetLocaleForCodePage(int Category, uint32_t CodePage) {
+#ifdef __APPLE__
+    switch (CodePage) {
+    case CP_ACP:
+      return setlocale(Category, "en_US.ISO8859-1");
+    case CP_UTF8:
+      return setlocale(Category, "en_US.UTF-8");
+    default:
+      return nullptr;
+    }
+#else
+    const char *utf8LocaleOptions[] = {"en_US.utf8", "en_US.UTF-8"};
```

Comment:
`"en_US.UTF-8"` seems like it might be the standard one, so why not put it first?  In fact, do we need any other option?

Looks like @damyanp already suggested that here:
https://github.com/microsoft/DirectXShaderCompiler/pull/6488#discussion_r1548651097